Lib/logging/__init__.py

```diff
@@ -1686,14 +1696,8 @@ def isEnabledFor(self, level):
         try:
             return self._cache[level]
         except KeyError:
-            _acquireLock()
-            if self.manager.disable >= level:
-                is_enabled = self._cache[level] = False
-            else:
-                is_enabled = self._cache[level] = level >= self.getEffectiveLevel()
-            _releaseLock()
-
-            return is_enabled
+            self._cache[level] = self._isEnabledForNoCache(level)
```

Comment:
The old version of the code locked the call to `self.getEffectiveLevel` against the update of the cache. The new version does not. This introduces a race condition:

1. Thread A consults the cache and gets a `KeyError`
2. Thread B also consults the cache and gets a `KeyError`
3. Thread A acquires the lock, reads the effective level, which is 1, and releases the lock
4. Thread C changes the effective level to 2
5. Thread B acquires the lock, reads the effective level, which is 2, and releases the lock
6. Thread B updates the cache to 2
7. Thread A updates the cache to 1

So now the effective level is 2 but the cached value is 1. This couldn't happen previously.

(Maybe this is harmless, I am not an expert on multi-threaded logging.)